api/app.rb

```diff
@@ -0,0 +1,111 @@
+require 'sinatra'
+require "sinatra/cross_origin"
+require 'byebug'
+require 'date'
+require 'restclient'
+require 'json'
+
+configure do
+  enable :cross_origin
+end
+
+before do
+  response.headers['Access-Control-Allow-Origin'] = '*'
+end
+
+def endpoint_call(base, month)
+  begin
+    # url = "https://fi7661d6o4.execute-api.eu-central-1.amazonaws.com/prod/indexes/#{params[:base_year]}/#{params[:current_month]}/"
+    # print url
+    url = "https://fi7661d6o4.execute-api.eu-central-1.amazonaws.com/prod/indexes/#{base}/#{month}"
+    response = RestClient.get url
+    intermediary_response = JSON.parse(response)
+    # puts pretty_response = JSON.pretty_generate(intermediary_response)
+    intermediary_response.keys
+    base_index = intermediary_response["index"]["MS_HLTH_IDX"]
+    return base_index
+  rescue RestClient::ExceptionWithResponse => e
+    p e.response
+  end
```

Comment:
Good to use a gem to perform your api call. However, here with a simple get request, `open-uri` already does the job 😉 
```suggestion
  JSON.parse(open("https://fi7661d6o4.execute-api.eu-central-1.amazonaws.com/prod/indexes/#{base}/#{month}").read)["index"]["MS_HLTH_IDX"]
rescue OpenURI::HTTPError => e
  p e.response
```

Reply:
Wow! Will look into `open-uri` more, this is awesome refactoring! Indeed, using tools that  already exist for things is something I need to do work on more instead of trying re-inventing the wheel.